### PR TITLE
[MAINTENANCE] Fix tests in tests/integration/profiling/rule_based_profiler/ and tests/render/renderer/

### PR DESCRIPTION
**Pull request by billdirks**
Some tests failed when run in different orders and this is my third PR to address this issue.
I recommend looking at the diff without whitespace changes since most of the test file is indented with a `try/finally` block  with very little functional changes.

This is for GREAT-1048.

Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

Changes proposed in this pull request:
-
-
-

After submitting your PR, CI checks will run and @cla-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [ ] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [ ] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [ ] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: tests/integration/profiling/rule_based_profiler/test_profiler_user_workflows.py [hunk 1]
```diff
@@ -1220,244 +1220,252 @@ def test_bobby_expect_column_values_to_be_between_auto_yes_default_profiler_conf
     default_profiler_config: Optional[
         RuleBasedProfilerConfig
     ] = expectation_impl.default_kwarg_values.get("profiler_config")
-    if default_profiler_config:
-        del expectation_impl.default_kwarg_values["profiler_config"]
 
-    assert "profiler_config" not in expectation_impl.default_kwarg_values
+    try:
```
Comment (billdirks) [inline tests/integration/profiling/rule_based_profiler/test_profiler_user_workflows.py:1224]: This change introduces a `try/finally` leaving the intermediate code unchanged except for 1 black formatting change.

File: tests/integration/profiling/rule_based_profiler/test_profiler_user_workflows.py [hunk 2]
```diff
@@ -1220,244 +1220,252 @@ def test_bobby_expect_column_values_to_be_between_auto_yes_default_profiler_conf
...
-                "parameter_builders": [
-                    {
-                        "name": "my_min_estimator",
-                        "class_name": "MetricMultiBatchParameterBuilder",
-                        "module_name": "great_expectations.rule_based_profiler.parameter_builder",
-                        "metric_name": "column.min",
-                        "metric_domain_kwargs": "$domain.domain_kwargs",
-                        "enforce_numeric_metric": True,
-                        "replace_nan_with_zero": True,
+        context: DataContext = (
```
Comment (billdirks) [inline tests/integration/profiling/rule_based_profiler/test_profiler_user_workflows.py:1230]: A black formatting change.

File: tests/integration/profiling/rule_based_profiler/test_profiler_user_workflows.py [hunk 3]
```diff
@@ -1220,244 +1220,252 @@ def test_bobby_expect_column_values_to_be_between_auto_yes_default_profiler_conf
...
+        assert result.expectation_config["kwargs"] == {
+            "column": "fare_amount",
+            "min_value": -21.0,
+            "mostly": 1.0,
+            "result_format": "SUMMARY",
+            "include_config": True,
+            "auto": False,
+            "batch_id": "0808e185a52825d22356de2fe00a8f5f",
+        }
+    finally:
```
Comment (billdirks) [inline tests/integration/profiling/rule_based_profiler/test_profiler_user_workflows.py:1465]: This finally block is the new piece of functionality.

File: tests/test_fixtures/great_expectations_custom_notebooks.yml [hunk 4]
```diff
@@ -79,7 +79,7 @@ notebooks:
     suite_edit:
         class_name: SuiteEditNotebookRenderer
         module_name: great_expectations.render.renderer.suite_edit_notebook_renderer
-        custom_templates_module: notebook_assets.suite_edit
+        custom_templates_module: tests.test_fixtures.notebook_assets.suite_edit
```
Comment (billdirks) [inline tests/test_fixtures/great_expectations_custom_notebooks.yml:82]: Without the full path, a test loading the custom module would sometimes fail.

**Review by alexsherstinsky — CHANGES_REQUESTED** [review]
@billdirks Could you please walk me through the changes over a brief call?  As it stands, it is not clear to me what was not working with these tests and how they are related to the order of tests being run (the profiler tests should not be order dependent).  I am also seeing what appears to be changes in the expected results, which means I am either not reading the PR in the best way, or maybe we do not need these changes, so let us discuss together.  Thanks!

**Review by alexsherstinsky — APPROVED** [review]
LGTM (pending the tests passing in Azure) -- @billdirks thank you so much for doing this and for the conversation!

**Pull request merged**